chia/full_node/full_node.py

```diff
@@ -2133,75 +2133,64 @@ async def _can_accept_compact_proof(
-                    if (
-                        sub_slot.infused_challenge_chain is not None
-                        and sub_slot.infused_challenge_chain.infused_challenge_chain_end_of_slot_vdf == vdf_info
-                    ):
-                        new_proofs = dataclasses.replace(sub_slot.proofs, infused_challenge_chain_slot_proof=vdf_proof)
-                        new_subslot = dataclasses.replace(sub_slot, proofs=new_proofs)
-                        new_finished_subslots = block.finished_sub_slots
-                        new_finished_subslots[index] = new_subslot
-                        new_block = dataclasses.replace(block, finished_sub_slots=new_finished_subslots)
-                        break
-            if field_vdf == CompressibleVDFField.CC_SP_VDF:
-                if block.reward_chain_block.challenge_chain_sp_vdf == vdf_info:
-                    assert block.challenge_chain_sp_proof is not None
-                    new_block = dataclasses.replace(block, challenge_chain_sp_proof=vdf_proof)
-            if field_vdf == CompressibleVDFField.CC_IP_VDF:
-                if block.reward_chain_block.challenge_chain_ip_vdf == vdf_info:
-                    new_block = dataclasses.replace(block, challenge_chain_ip_proof=vdf_proof)
-            if new_block is None:
-                continue
-            async with self.db_wrapper.lock:
-                peak: Optional[BlockRecord] = self.blockchain.get_peak()
-                assert peak is not None
-                if new_block.header_hash == peak.header_hash or peak.height - new_block.height < 5:
-                    continue
-                main_chain_hash: Optional[bytes32] = self.blockchain.height_to_hash(new_block.height)
-                assert main_chain_hash is not None
-                in_main_chain: bool = main_chain_hash == new_block.header_hash
-                try:
-                    await self.block_store.db_wrapper.begin_transaction()
-                    await self.block_store.add_full_block(new_block.header_hash, new_block, block_record, in_main_chain)
-                    await self.block_store.db_wrapper.commit_transaction()
-                    replaced = True
-                except BaseException as e:
-                    await self.block_store.db_wrapper.rollback_transaction()
-                    self.log.error(
-                        f"_replace_proof error while adding block {block.header_hash} height {block.height},"
-                        f" rolling back: {e} {traceback.format_exc()}"
-                    )
-                    raise
-        return replaced
+
+        block = await self.block_store.get_full_block(header_hash)
+        if block is None:
+            return False
+
+        new_block = None
+
+        if field_vdf == CompressibleVDFField.CC_EOS_VDF:
+            for index, sub_slot in enumerate(block.finished_sub_slots):
+                if sub_slot.challenge_chain.challenge_chain_end_of_slot_vdf == vdf_info:
```

Comment:
i think we should error in cases where the vdf_info doesn't match 
since now we pull the block from the db by hash basically what we have here is the same block with different vdf_info

not suggesting we fix that in this pr